staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/interface.go

```diff
@@ -95,3 +98,23 @@ type Filter interface {
 	// CompilationErrors returns a list of errors from the compilation of the evaluator
 	CompilationErrors() []error
 }
+
+// EvaluatorCompiler contains a function to assist with converting types and values to/from CEL-typed values.
+type EvaluatorCompiler interface {
+	// CompileEvaluator is used for the cel expression compilation
+	CompileEvaluator(expression ExpressionAccessor, optionalDecls OptionalVariableDeclarations, envType environment.Type) Evaluator
+}
+
+// Evaluator contains the result of compiling a CEL expression.
+// It expects the inbound object to already have been converted to the version expected
+// by the underlying CEL code (which is indicated by the match criteria of a policy definition).
+// versionedParams may be nil.
+type Evaluator interface {
```

Comment:
it's unclear to me how this differs from `CompilationResult`. One of these is likely badly named. `CompilationResult`'s go doc talks about validation. Is that the difference?

Reply:
Next question: `Filter` is used for validation, `Evaluator` is used for mutation. Neither is really apparent from the name, and both have **exactly** the same go doc. These two concepts need better names and better docs. How do they differ? Is a `Filter` actually a `Validator`? What does it filter after all?

Reply:
Or `ValidatingEvaluator` and `MutatingEvaluator`.

Reply:
I like `MutatingEvaluator`.  I'd like to use `ConditionEvaluator` instead of `ValidatingEvaluator` since it is used for both validations and match conditions. I'll clarify that detail in the godoc.